src/bakery.js

```diff
@@ -28,7 +28,10 @@ const ERR_DISCHARGE_REQUIRED = "macaroon discharge required";
 const ERR_INTERACTION_REQUIRED = "interaction required";
 
 let TextDecoder;
-if (!TextDecoder) {
+if (typeof window !== "undefined" && window && window.TextDecoder) {
```

Comment:
`TextDecoder` was always undefined and `util.TextDecoder` was used instead of the browser implementation when running outside of node.